sdk/identity/azure-identity/tests/test_vscode_credential_async.py

```diff
@@ -0,0 +1,70 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import sys
+import pytest
+from azure.core.credentials import AccessToken
+from azure.identity._internal.user_agent import USER_AGENT
+from azure.identity import CredentialUnavailableError
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+from helpers import build_aad_response, mock_response, Request
+from helpers_async import async_validating_transport, AsyncMockTransport, wrap_in_future
+from unittest import mock
+from azure.identity.aio._credentials.vscode_credential import VSCodeCredential
+
+
+@pytest.mark.asyncio
+async def test_no_scopes():
+    """The credential should raise ValueError when get_token is called with no scopes"""
+
+    credential = VSCodeCredential()
+    with pytest.raises(ValueError):
+        await credential.get_token()
+
+
+@pytest.mark.asyncio
+async def test_policies_configurable():
+    policy = mock.Mock(spec_set=SansIOHTTPPolicy, on_request=mock.Mock())
+
+    async def send(*_, **__):
+        return mock_response(json_payload=build_aad_response(access_token="**"))
+
+    with mock.patch('azure.identity.aio._credentials.vscode_credential.get_credentials', return_value="VALUE"):
+        credential = VSCodeCredential(policies=[policy], transport=mock.Mock(send=send))
+        await credential.get_token("scope")
+        assert policy.on_request.called
+
+
+@pytest.mark.asyncio
+async def test_user_agent():
+    transport = async_validating_transport(
+        requests=[Request(required_headers={"User-Agent": USER_AGENT})],
+        responses=[mock_response(json_payload=build_aad_response(access_token="**"))],
+    )
+
+    with mock.patch('azure.identity.aio._credentials.vscode_credential.get_credentials', return_value="VALUE"):
```

Comment:
I've started using `__module__` to make patches easier to maintain. You can also omit the `return_value` when it isn't important (the patch returns a `Mock` by default):
```suggestion
    with mock.patch(VSCodeCredential.__module__ + ".get_credentials"):
```